File: app/src/main/java/org/wikipedia/views/WrapContentViewPager.java
package org.wikipedia.views;

import android.content.Context;
import android.util.AttributeSet;
import android.view.View;

import com.duolingo.open.rtlviewpager.RtlViewPager;

public class WrapContentViewPager extends RtlViewPager {
    private int maxHeight = Integer.MAX_VALUE;

    public WrapContentViewPager(Context context) {
        super(context);
    }

    public WrapContentViewPager(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public void setMaxHeight(int height) {
        maxHeight = height;
    }

    @Override protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        try {
            int height = 0;
            for (int i = 0; i < getChildCount(); i++) {
                View child = getChildAt(i);
                child.measure(widthMeasureSpec, MeasureSpec.makeMeasureSpec(0, MeasureSpec.UNSPECIFIED));
                int h = child.getMeasuredHeight();
                if (h > height) {
                    height = h;
                }
            }
            if (height != 0) {
                heightMeasureSpec = MeasureSpec.makeMeasureSpec(Math.min(height, maxHeight), MeasureSpec.EXACTLY);
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
        super.onMeasure(widthMeasureSpec, heightMeasureSpec);
    }
}

File: app/src/main/java/org/wikipedia/views/WrapContentViewPager.kt
package org.wikipedia.views

import android.content.Context
import android.util.AttributeSet
import com.duolingo.open.rtlviewpager.RtlViewPager
import kotlin.math.min

class WrapContentViewPager @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null): RtlViewPager(context, attrs) {

    private var maxHeight = Int.MAX_VALUE

    fun setMaxHeight(height: Int) {
        maxHeight = height
    }

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        var newHeightMeasureSpec = heightMeasureSpec
        try {
            var height = 0
            for (i in 0 until childCount) {
                val child = getChildAt(i)
                child.measure(widthMeasureSpec, MeasureSpec.makeMeasureSpec(0, MeasureSpec.UNSPECIFIED))
                val h = child.measuredHeight
                if (h > height) {
                    height = h
                }
            }
            if (height != 0) {
                newHeightMeasureSpec = MeasureSpec.makeMeasureSpec(min(height, maxHeight), MeasureSpec.EXACTLY)
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
        super.onMeasure(widthMeasureSpec, newHeightMeasureSpec)
    }
}
